### ISSUE TITLE
import cv2 raise importerror. MacOS

### ISSUE THREAD

**Issue by heisenger** [opencv/opencv#13991]
```
ImportError: dlopen(/Users/julian/.virtualenvs/cv/lib/python3.6/site-packages/cv2.so, 2): Symbol not found: __ZN2cv3dnn14dnn4_v201809175Layer10initHalideERKNSt3__16vectorINS_3PtrINS1_14BackendWrapperEEENS3_9allocatorIS7_EEEE
  Referenced from: /Users/julian/.virtualenvs/cv/lib/python3.6/site-packages/cv2.so
  Expected in: /usr/local/lib/libopencv_dnn.4.0.dylib
 in /Users/julian/.virtualenvs/cv/lib/python3.6/site-packages/cv2.so
```

I have built it from source on a macOS system - also tried disabling SIP as others suggested - but it doesnt seem to resolve the issue. Does anybody know why?
Thanks

**Comment by alalek** [comment]
Check modification date of mentioned files.
Perhaps you have old files due running of "make install" without "make uninstall" before upgrade.

**Comment by heisenger** [comment]
Thankyou! It resolved the issue I have

Just want to check 2 more points, as I think I am doing it wrong: 
1) when I run make uninstall in the build directory - it doesnt seem to delete the installed files, so I removed files with *opencv* myself. 
2) I deleted the folder /usr/local/python/cv2, thinking that if I rerun the build and installation, it will re-create this directory. Turns out it does not - and i actually manually  recreated this directory and placed the cv2.so myself. What did i do wrong? (i followed https://www.pyimagesearch.com/2018/08/17/install-opencv-4-on-macos/) 

Many thanks again
